GoogleCloudPlatform/microservices-demo
# Port `NetworkPolicy` flavour to Kustomize

### Background 
* This is part of https://github.com/GoogleCloudPlatform/microservices-demo/issues/536.
* We want create Kustomize `Component`s for each flavour of Online Boutique — where possible.
* This pull-request is specifically for [/docs/network-policies](https://github.com/GoogleCloudPlatform/microservices-demo/tree/main/docs/network-policies).

### Change Summary
* This pull-request ports over the flavour with `NetworkPolicy` resources to Kustomize.

### Testing Procedure

1. Check out the source branch:
```
git checkout nimjay-kustomize-network-policy
```

2. Uncomment the `network-policy` Kustomize Component inside `kustomize/kustomization.yaml`.
```
sed -i '' 's/# - components\/network-policy/- components\/network-policy/' kustomize/kustomization.yaml
```

3. Check that Kustomize builds correctly.
```
# This is the same as "kustomize build".
kubectl kustomize ./kustomize/
```

4. Check that the Kustomize build output includes the `NetworkPolicy` resources.
```
kubectl kustomize ./kustomize/ | grep NetworkPolicy
```
You'll notice that the `NetworkPolicy` resources are included in the Kustomize build.

### Related PRs or Issues 
* Here's the pull-request where we added these `NetworkPolicy` samples: https://github.com/GoogleCloudPlatform/microservices-demo/pull/778
